packages/core/src/data-editor/data-editor.tsx

```diff
@@ -2091,6 +2091,59 @@ const DataEditorImpl: React.ForwardRefRenderFunction<DataEditorRef, DataEditorPr
 
     const [scrollDir, setScrollDir] = React.useState<GridMouseEventArgs["scrollEdge"]>();
 
+    // Adjust selection to include latest cell in visible region.
+    React.useEffect(() => {
```

Comment:
I'm not sure this is right mechanism to do this. The easiest thing to do would be to find a way to reasonably re-emit the mouse move, triggering a new onItemHovered event as scroll is going on. The state update then would become a natural consequene of the artificial mousemove events.

Reply:
Makes sense. Most of the logic for adjusting the selection in this useEffect I just took from the `adjustSelection` function, but emitting artificial mousemove events is a better way of code re-usability and correctness.

Reply:
PR now triggers onItemHovered by listening to the `useAutoscroll` callback.